code/datums/mind.dm

```diff
@@ -800,7 +800,8 @@
 				qdel(flash)
 				take_uplink()
 				var/fail = 0
-				fail |= !current.mind.add_antag_datum(/datum/antagonist/traitor)
+				var/datum/antagonist/traitor/T = current

```

Comment:
don't think you should be passing current as an argument here, either just use new() or use has_antag_datum(/datum/antagonist/traitor) as that returns the minds traitor datum